### PR TITLE
Map phone number lookup results to their native rooms

### PR DESCRIPTION
**Pull request by dbkr**
When dialing a phone number, also look to see if there's a corresponding
native user for the resulting user, and if so, go to the native room
for that user.

Also fix as bug where we were checking the wrong field to check
for native/virtual support: oops.

Move phone number dial handling into CallHandler rather than the DialPad component.

### CODE REVIEW

File: src/dispatcher/actions.ts [hunk 1]
```diff
@@ -100,6 +100,14 @@ export enum Action {
      */
     OpenDialPad = "open_dial_pad",
 
+    /**
+     * Dial the phone number in the payload
+     * payload: {
+     *     number: <phone number>,
+     * }
```
Comment (turt2live) [inline src/dispatcher/actions.ts:107]: this should be interfaced like the surrounding actions

File: test/CallHandler-test.ts [hunk 2]
```diff
@@ -134,38 +161,34 @@ describe('CallHandler', () => {
...
-                case REAL_ROOM_ID:
-                    return realRoom;
-                case MAPPED_ROOM_ID:
-                    return mappedRoom;
-                case MAPPED_ROOM_ID_2:
-                    return mappedRoom2;
-            }
-        };
+        dis.dispatch({
+            action: 'dial_number',
```
Comment (turt2live) [inline test/CallHandler-test.ts:175]: the action enum would be good to use here

**Review by turt2live — CHANGES_REQUESTED** [review]
generally looks fine, just some stylistic comments

**Comment by chagai95** [comment]
@tomtom-78 @chagai95 are waiting for this to be merged - hope we get an email now when this is merged 😅

**Review by turt2live — APPROVED** [review]

**Pull request merged**